xonsh/events.py

```diff
@@ -0,0 +1,113 @@
+"""
+Events for xonsh.
+
+In all likelihood, you want builtins.__xonsh_events__
+
+The best way to "declare" an event is something like::
+
+    __xonsh_events__.on_spam.__doc__ = "Comes with eggs"
```

Comment:
It forces the event to exist, and it sets the docstring for it. (Except that in this version, nothing will use that docstring. I'm working on it.)